src/target/linux/writemem.rs

```diff
@@ -0,0 +1,304 @@
+use nix::{sys::ptrace, unistd::Pid};
+use std::{cmp, marker::PhantomData, mem, slice};
+
+const WORD_SIZE: usize = mem::size_of::<usize>();
+
+/// Allows to write data to different locations in debuggee's memory as a single operation.
+/// This implementation can select different strategies for different memory pages.
+pub struct WriteMemory<'a> {
+    pid: Pid,
+    write_ops: Vec<WriteOp>,
+    _marker: PhantomData<&'a ()>,
+}
+
+impl<'a> WriteMemory<'a> {
+    pub(super) fn new(pid: Pid) -> Self {
+        WriteMemory {
+            pid,
+            write_ops: Vec::new(),
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn write<T: ?Sized>(mut self, val: &'a T, remote_base: usize) -> Self {
+        self.write_ops.push(WriteOp {
+            remote_base,
+            source_len: mem::size_of_val(val),
+            source_ptr: val as *const T as *const libc::c_void,
+        });
+        self
+    }
+
+    /// Executes the memory write operation.
+    pub unsafe fn apply(self) -> Result<(), Box<dyn std::error::Error>> {
+        write_process_vm(self.pid, &self.write_ops)
+    }
+}
+
+/// A single memory write operation.
+#[derive(Debug, PartialEq, Eq)]
+pub(crate) struct WriteOp {
+    // Remote destation location.
+    remote_base: usize,
+    // Pointer to a source.
+    source_ptr: *const libc::c_void,
+    // Size of `source_ptr`.
```

Comment:
```suggestion
    /// Remote destation location.
    remote_base: usize,
    /// Pointer to a source.
    source_ptr: *const libc::c_void,
    /// Size of `source_ptr`.
```

These show up when using `--document-private-items`.